utils/src/main/java/org/web3j/utils/Numeric.java

```diff
@@ -47,13 +47,25 @@ public static BigInteger decodeQuantity(String value) {
         if (!isValidHexQuantity(value)) {
             throw new MessageDecodingException("Value must be in format 0x[1-9]+[0-9]* or 0x0");
         }
+
         try {
-            return new BigInteger(value.substring(2), 16);
+            return new BigInteger(parsePaddedHexQuantity(value), 16);
         } catch (NumberFormatException e) {
             throw new MessageDecodingException("Negative ", e);
         }
     }
 
+    private static String parsePaddedHexQuantity(String value) {
```

Comment:
I suggest to make simpler solution 
```
public static BigInteger parsePaddedNumberHex(String value) {
        String numWithoutLeadingZeros = cleanHexPrefix(value).replaceFirst("^0+", "");
        return numWithoutLeadingZeros.isEmpty() ? BigInteger.ZERO : new BigInteger(numWithoutLeadingZeros, 16);
    }
```

Reply:
Yeah seems Regex could have been a better approach here for replacing leading zeros

Reply:
Yeah coded purely in java for simple solution, but can make it more simpler using regex